### ISSUE TITLE
Edge browser - XHR request causes "ProgressEvent" error when debugging localhost to remote server

### ISSUE THREAD

**Issue by JamesHough** [angular/angular#34493]
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/common/http

### Is this a regression?
Unknown. I first noticed this in one of the latest 8.x releases.  I haven't checked all of the previous versions, but I suspect that it was working in 7.x (or an earlier Edge browser version) when I was more actively developing this project.

### Description
When posting to a remote server from localhost, an error is thrown in **Edge browser** but no other browser does this (tested on Edge, Firefox, Chrome).

The error itself appears to be invalid because the error object is a `ProgressEvent`. This may be because XHR on Edge browser has a different progress event being raised that other browsers don't have and that unexpected event causes an error.

![image](https://user-images.githubusercontent.com/12199027/71175035-2ffc4e80-226f-11ea-8e6a-fe771cf6e990.png)

If you have the server also running on localhost then it succeeds in all browsers  (Edge, Firefox, Chrome).

Note: _this isn't server-side CORS_ because other browsers don't get the error when calling the same server.  Also, in the example below, pure HTML/JS does not generate the error.

## 🔬 Minimal Reproduction
This repo contains a demo Angular project and a Node Express server to accept uploaded files (it does nothing with the uploaded files, just returns 200 and a fixed message).

https://github.com/JamesHough/angular-edge-progressevent-error

Run `npm install` inside the root folder **and also** inside the "node" folder which has the Express REST api for testing.

`npm start` will start the local REST server and the Angular app. _However_ you must copy the REST service to a different server and host it in Node in order to cause the problem to occur.  When you move it to a remote server, you will need to edit `any-old.service.ts` and also edit `index.html` to set the right remote server to use for testing.

Note: This sample includes both an Angular solution and a plain HTML/JS solution that will both upload to the same Node Express service.  The HTML/JS example uses XHR and works on all browsers (Edge, Firefox, Chrome)

## 🔥 Exception or Error
<pre><code>{
  error: ProgressEvent,
  headers: Object,
  message: "Http failure response for http://remote-server-name:3000/api/excel-reader-api/api/ReadColumnHeaders: 0 Unknown Error",
  name: "HttpErrorResponse",
  ok: false,
  status: 0,
  statusText: "Unknown Error",
  url: "http://remote-server-name:3000/api/excel-reader-api/api/ReadColumnHeaders"
}</code></pre>

## 🌍  Your Environment
**Angular Version:**
<pre><code>Angular CLI: 8.3.20
Node: 12.5.0
OS: win32 x64
Angular: 8.2.14
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.20
@angular-devkit/build-angular     0.803.20
@angular-devkit/build-optimizer   0.803.20
@angular-devkit/core              8.3.20
@angular-devkit/schematics        8.3.20
@angular/cli                      8.3.20
@ngtools/webpack                  8.3.20
@schematics/angular               8.3.20
@schematics/update                0.803.20
rxjs                              6.4.0
typescript                        3.5.3
webpack                           4.39.2</code></pre>

**Anything else relevant?**
This only happens on the Edge browser when running on localhost and connecting to a remote server.  I have Edge version 18 installed but it may have been working on earlier Edge browser versions (or in earlier Angular versions).
<code><pre>Microsoft Edge 44.18362.449.0
Microsoft EdgeHTML 18.18363</code></pre>

**Comment by JamesHough** [comment]
Additional info that might be helpful; while debugging and isolating in which scenarios this problem occurs, I noticed that the error comes straight from the `BrowserXhr` component (which I assume is wrapping the native browser XHR provider) and when this error is raised at this low level, it's already of type `ProgressEvent` and has all the same information that Angular bubbles up to the caller.

`const xhr = this.xhrFactory.build();`
...
`xhr.addEventListener('error', onError);`
...
![image](https://user-images.githubusercontent.com/12199027/71179519-06482500-2279-11ea-8b41-9654a5fdfeee.png)

**Comment by JamesHough** [comment]
To answer my own question above; yup, the plain browser XHR is being returned.

<code<pre>@Injectable()

export class BrowserXhr {

  constructor() {}

  build(): any { return <any>(new XMLHttpRequest()); }

}</pre></code>

**Comment by JamesHough** [comment]
Okay, I recreated the same condition in pure JavaScript.  Adding the event handlers highlighted that the same issue was occurring.  I had a bug in my sample which I've updated now - but after I fixed that bug it was the same.  For Chrome and Firefox it works but for Edge it does not.  _It's only when the web page is on localhost and the server is a remote server._ 

We can close this because it's not Angular.  I will get in touch with Microsoft.

![image](https://user-images.githubusercontent.com/12199027/71234496-d4c76c00-2301-11ea-9131-1a283ec67bbe.png)
